### PR TITLE
feat: use typescript instead of babel as typescript compiler in the server

### PR DESCRIPTION
**Pull request by yimingjfe**
# PR Details

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] I have updated changeset
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: tests/integration/api-service-koa/package.json [hunk 1]
```diff
@@ -41,20 +41,22 @@
     "@modern-js/runtime": "workspace:*",
     "koa": "^2.13.4",
     "react": "^17.0.1",
-    "react-dom": "^17.0.1"
+    "react-dom": "^17.0.1",
+    "ts-node": "^10.1.0",
+    "tsconfig-paths": "3.14.1"
```
Comment (chenjiahan) [inline tests/integration/api-service-koa/package.json:46]: 可以用 @modern-js/utils 里预打包的 tsconfig-paths

File: packages/server/server/package.json [hunk 2]
```diff
@@ -41,6 +41,7 @@
     "@modern-js/prod-server": "workspace:*",
     "@modern-js/server-utils": "workspace:*",
     "@modern-js/utils": "workspace:*",
+    "@modern-js/webpack": "workspace:*",
```
Comment (chenjiahan) [inline packages/server/server/package.json:44]: 这里为啥加上了 @modern-js/webpack (之前刚去掉)

File: packages/server/server/package.json [hunk 3]
```diff
@@ -57,9 +58,15 @@
     "@types/node": "^14",
     "@types/ws": "^7.4.7",
     "jest": "^27",
+    "ts-node": "^10.1.0",
+    "tsconfig-paths": "3.14.1",
```
Comment (chenjiahan) [inline packages/server/server/package.json:62]: ts-node 同时存在于 dependencies 和 peerDependencies 里的原因是啥

File: tests/integration/api-service-koa/package.json (hunk 1 above)
Comment (yimingjfe) [reply tests/integration/api-service-koa/package.json:46]: 现在是把 tsconfig-paths 作为用户自己的 devDependencies 了，这里是集测中的项目，应该是加到  devDependencies，我修改下

File: packages/server/server/package.json (hunk 2 above)
Comment (yimingjfe) [reply packages/server/server/package.json]: 应该是误加了，我删一下~

**Review by zllkjc — DISMISSED** [review]

**Review by caohuilin — APPROVED** [review]

**Pull request merged**